### DISCUSSION TITLE
Google Dynamorio Traces

### DISCUSSION THREAD

**Discussion by jogli5er** [ChampSim/ChampSim#374] [Q&A] [answered]
Has anyone used the [Google Dynamorio Traces](https://dynamorio.org/google_workload_traces.html) under ChampSim or written a specific trace reader/trace translator for it? The information available should be sufficient to generate ChampSim traces, but I would like to avoid duplicating the work. Is someone aware of such an implementation?

Best,
Roman

**Comment by ngober** [comment, marked as answer]
There was a lot of discussion when the traces were initially released, but it seems to have died down. I think the consensus was that the traces do not include any timing information. ChampSim traces include source and destination registers (or manufactures them, see #372) and from that generates timing information by simulating an out-of-order core. @djimeneth might know more, because he actually reached out to the team that released them.

**Reply by jogli5er** [reply]
Thank you for the quick reply and the pointers!

**Reply by jogli5er** [reply]
Following up on the discussion in the [DynamoRIO mailing list](https://groups.google.com/g/dynamorio-users/c/TU2wa_Wvqbs), I'd be happy to work towards a converter from DynamoRIO to ChampSim traces. There are a few things:

1. I took a look at #372. The same logic can be applied to the DynamoRIO traces for quite a few cases.
2. My current focus lies on the Google traces. I would start with those, even though they represent just a subset of a full trace. However, extending such a first draft should be straightforward since it is a proper subset.

You mentioned some discussions with respect to the Google traces when they were released. Were those discussions on the old mailing list of ChampSim?

**Reply by ngober** [reply]
These were mostly personal communications. I'm not certain there's a record of them.

It looks like @hrishi-06 has something that could be useful. Take a look at that pull request, when it comes in, and see if it meets your needs.

**Comment by hrishi-06** [comment]
> Has anyone used the [Google Dynamorio Traces](https://dynamorio.org/google_workload_traces.html) under ChampSim or written a specific trace reader/trace translator for it? The information available should be sufficient to generate ChampSim traces, but I would like to avoid duplicating the work. Is someone aware of such an implementation?
> 
> Best, Roman

I have written code for the same. I will be happy to share it with you.

**Reply by ngober** [reply]
I'm excited to hear that. Would you please submit it as a pull request so it can be reviewed?

**Reply by jogli5er** [reply]
That'd be great!

**Reply by hrishi-06** [reply]
I have created a pull request [here](https://github.com/ChampSim/ChampSim/pull/379#issue-1867286602)
